frontend/app/.server/routes/helpers/protected-renew-route-helpers.ts

```diff
@@ -196,24 +200,31 @@ export function clearProtectedRenewState({ params, session }: ClearStateArgs) {
 interface StartArgs {
   id: string;
   session: Session;
+  appContainer: AppContainerProvider;
 }
 
 /**
  * Starts protected renew state.
  * @param args - The arguments.
  * @returns The initial protected renew state.
  */
-export function startProtectedRenewState({ id, session }: StartArgs) {
+export async function startProtectedRenewState({ id, session, appContainer }: StartArgs) {
   const log = getLogger('protected-renew-route-helpers.server/startProtectedRenewState');
   const parsedId = idSchema.parse(id);
+  const sessionName = getSessionName(parsedId);
+
+  const clientApplicationService = appContainer.get(TYPES.domain.services.ClientApplicationService);
+  const clientApplication = await clientApplicationService.findClientApplicationBySin(session.get('userInfoToken').sin);
+  invariant(clientApplication, 'Expected client application to be defined');
```

Comment:
We can redirect to the `/data-unavailable` route if no client application is found.